lib/katello/tasks/reindex.rake

```diff
@@ -1,6 +1,38 @@
 namespace :katello do
-  task :reindex => ["environment", "katello:reset_backends:elasticsearch"]  do
-    User.current = User.first #set a user for orchestration
+  def report_bad_objects(bad_objects_exception_hash, model_name)
+    User.current = User.anonymous_admin
+    puts "The following #{model_name} items could not be indexed due to various reasons."
+    bad_objects_exception_hash.each do |object, exception|
+      puts "Object: #{object.inspect}"
+      puts "Exception: #{exception.message}"
+      if object.is_a? Katello::Repository
+        notes = []
+        notes << "Pulp Repository #{object.pulp_id} was not found." if object.pulp_repo_facts.nil?
+        notes << "Candlepin Content was not available for #{object.name}." if object.content.nil?
```

Comment:
found that if there's exception in this function, reindex would stop
for example  `object.content` threw this trace for me.

https://gist.github.com/dustints/06ccccd31399f2974605